numpy/core/fromnumeric.py

```diff
@@ -1827,6 +1827,11 @@ def sum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
         `ndarray`, however any non-default value will be.  If the
         sub-class' method does not implement `keepdims` any
         exceptions will be raised.
+    initializer : scalar, optional
+        The initializer for this maximum. Can also be used to compute the
+        minimum of an empty slice.
```

Comment:
`max`? `min`? This is `sum`

I'd argue that this isn't useful enough to include in the public API here, or in any of these functions. It's niche enough that we probably want the user calling `ufunc.reduce` directly

Reply:
Hmm. I went to a lot of effort to put it in. :-( Anyway, I'd argue it should stay for the NaN-skipping functions because there's no alternative there.

Reply:
Additionally, it's useful for specifying the value of an All-NaN slice reduction (although I guess you could just do this):

```python
x_max = np.nanmax(x, ...)
x[np.isnan(x)] = value
```

Reply:
Here's what I propose:

* `np.some_ufunc.reduce(..., initializer=...)`
* `np.sum(...)`
* `np.prod(...)`
* `np.min(..., default=...)` - forward to initializer, match the py3 API
* `np.nanmin(..., default=...)` as above

Reply:
I take it back, `default` and `initializer` mean different things. `max([0], default=10)` is different to `np.max([0], initializer=10)`

Reply:
Exactly, as in the discussion above. But to keep uniformity, we should probably either keep it or remove it everywhere. I prefer users should not have to remember/recall which reductions had this and which didn't, they should just instinctively go for either the reduction or `ufunc.reduce`.

I guess to cater for the `nanmax`, etc. case, we could ask users to reshape, add an extra slice, then undo, but that would be excessive.

Reply:
nanmax absolutely should have an `initializer` argument, since it needs it to deal with all-nan slices of object dtypes. I'm just not sure they're useful on `sum` and `prod`, but maybe leave them in for now, and consult the list.

Either way, my initial remark here is pointing out that this docstring is copied-and-pasted incorrectly

Reply:
That's fixed. I see you noticed it was a copypasta.